knotx-knot/knotx-knot-handlebars/src/main/java/io/knotx/knot/templating/impl/HandlebarsKnotProxyImpl.java

```diff
@@ -69,6 +92,41 @@ public HandlebarsKnotProxyImpl(HandlebarsKnotConfiguration configuration) {
     });
   }
 
+  private Template template(Fragment fragment) {
+    try {
+      byte[] cacheKeyBytes = digest.digest(fragment.content().getBytes(StandardCharsets.UTF_8));
+      String cacheKey = new String(cacheKeyBytes);
+
+      return cache.get(cacheKey, () -> {
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Compiles Handlebars fragment [{}]",
+              StringUtils
+                  .abbreviate(fragment.content().replaceAll("[\n\r\t]", ""),
+                      DEBUG_MAX_FRAGMENT_CONTENT_LOG_LENGTH));
+        }
+        return handlebars.compileInline(FragmentContentExtractor.unwrapContent(fragment));
+      });
+    } catch (ExecutionException e) {
+      LOGGER.error("Could not compile fragment [{}]", fragment.content(), e);
+      throw new IllegalStateException(e);
+    }
+  }
+
+  private String evaluate(Template template, JsonObject context) {
+    if (LOGGER.isTraceEnabled()) {
+      LOGGER.trace("Applying context [{}] to template [{}]!", context, template.text());
```

Comment:
Consider abbreviation of template.text()

Reply:
fixed